modules/juce_core/native/juce_win32_Threads.cpp

```diff
@@ -445,15 +482,15 @@ class ChildProcess::ActiveProcess
 
             if (available == 0)
             {
-                if (! isRunning())
+                if (total != 0 || ! isRunning())

```

Comment:
this is to produce the output faster on windows and avoid spinning if we alread have characters. this will make the shell output flow more fluently instead of getting out very slow and clunky on processes that produce lot of output (like build systems).